### PR TITLE
291 note modify axios

### PR DESCRIPTION
**Pull request by DutchBen**
Created a ProcessDetail component to be used in multiple places

### CODE REVIEW

File: packages/orchestrator-ui-components/src/components/WFOForms/AutoFieldLoader.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    props: GuaranteedProps<unknown> & Record<string, unknown>,
+    uniforms: Context<Record<string, unknown>>,
+) {
+    const { allowedValues, fieldType, field } = props;
+    const { format } = field;
+
+    switch (fieldType) {
+        case Number:
+            switch (format) {
+                case 'imsPortId':
```
Comment (pboers1988) [inline packages/orchestrator-ui-components/src/components/WFOForms/AutoFieldLoader.tsx:24]: This is a SURF specific field.

File: packages/orchestrator-ui-components/src/components/WFOForms/AutoFieldLoader.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                    return NumField;
+                case 'imsNodeId':
```
Comment (pboers1988) [inline packages/orchestrator-ui-components/src/components/WFOForms/AutoFieldLoader.tsx:26]: SURF specific field

File: apps/wfo-ui/package.json [hunk 3]
```diff
@@ -8,7 +8,8 @@
         "tsc": "tsc --noEmit",
         "build": "next build",
         "start": "next start",
-        "lint": "next lint"
+        "lint": "next lint",
+        "reset": "rm -rf node_modules"
```
Comment (ricardovdheijden) [inline apps/wfo-ui/package.json:12]: What is the usecase for this?

File: apps/wfo-ui/pages/forms.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { EuiHorizontalRule, EuiPageHeader, EuiSpacer } from '@elastic/eui';
+import React from 'react';
+import CreateForm from '@orchestrator-ui/orchestrator-ui-components/src/components/WFOForms/CreateForm';
+
+export function FormsPage() {
```
Comment (ricardovdheijden) [inline apps/wfo-ui/pages/forms.tsx:5]: Just to check: we do want this one for now in the MVP branch?

File: apps/wfo-ui/pages/start-workflow/[workflowName].tsx [hunk 5]
```diff
@@ -1,15 +1,20 @@
...
 
-    const processName = Array.isArray(workflowNameQueryParameter)
-        ? workflowNameQueryParameter[0]
-        : workflowNameQueryParameter;
-
-    return <h1>Start workflow: {processName}</h1>;
+    if (workflowName && productId) {
+        return (
+            <WFOStartWorkflowPage
+                workflowName={workflowName as string}
```
Comment (ricardovdheijden) [inline apps/wfo-ui/pages/start-workflow/[workflowName].tsx:12]: The router is able to return an array, i think it is better to use the mechanism that is removed (Array.isArray) -- maybe hide it in a util file

File: packages/orchestrator-ui-components/src/api/index.ts [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    };
+    productById = (productId: string): Promise<ProductDefinition> => {
+        return this.fetchJson(`products/${productId}`);
+    };
+    cimStartForm = (
+        formKey: string,
+        userInputs: object[],
+    ): Promise<{ id: string }> => {
+        return this.postPutJson(
+            `surf/forms/${formKey}`,
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/api/index.ts:124]: There are multiple urls or partial URLs here. Maybe move them to the top of the file as a const of some other convenient place.

File: packages/orchestrator-ui-components/src/components/WFOForms/CreateForm.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import { Form, FormNotCompleteResponse } from '../../types/forms';
+import UserInputFormWizard from './UserInputFormWizard';
+import { apiClient } from '../../api';
+
+interface IProps {
+    /* eslint-disable  @typescript-eslint/no-explicit-any */
+    preselectedInput?: any;
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/components/WFOForms/CreateForm.tsx:23]: preselectedInput can be anything, but how about unknown or a key-value kind of type?

File: packages/orchestrator-ui-components/src/components/WFOForms/CreateForm.tsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+export default function CreateForm(props: IProps) {
+    const { preselectedInput, formKey, handleSubmit } = props;
+    const [form, setForm] = useState<Form>({});
+    const { stepUserInput, hasNext } = form;
+
+    const submit = useCallback(
+        (userInputs: object[]) => {
+            return apiClient.cimStartForm(formKey, userInputs).then((form) => {
+                console.log('Submit {formkey} =', formKey);
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/components/WFOForms/CreateForm.tsx:36]: FYI. I noticed multiple console logs/debugs in the PR.

File: packages/orchestrator-ui-components/src/api/axios.ts [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import axios, { AxiosRequestConfig } from 'axios';
+
+const apiPath = 'https://orchestrator.dev.automation.surf.net/api';
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/api/axios.ts:18]: The api path is also available in the config-context.

File: apps/wfo-ui/pages/forms.tsx (hunk 4 above)
Comment (DutchBen) [reply apps/wfo-ui/pages/forms.tsx:5]: AFAIK. It's fine, it shows an example implementation

File: apps/wfo-ui/pages/start-workflow/[workflowName].tsx (hunk 5 above)
Comment (DutchBen) [reply apps/wfo-ui/pages/start-workflow/[workflowName].tsx:12]: Anyway.. util functions is fine aswell. Will implement

File: packages/orchestrator-ui-components/src/api/axios.ts [hunk 10]
```diff
@@ -0,0 +1,38 @@
+/*
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/api/axios.ts:1]: The usage of the v1 forms code in this project is considered temporary, I would like to suggest to move this axios and index files into the WFOForms folder to have all the temporary code in 1 place. We would not plan to use this anywhere else in the project anyway

File: packages/orchestrator-ui-components/src/components/WFOForms/AutoFieldLoader.tsx (hunk 2 above)
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/components/WFOForms/AutoFieldLoader.tsx:26]: This is won't fix for the context of this PR

File: packages/orchestrator-ui-components/src/components/WFOForms/AutoFieldLoader.tsx (hunk 1 above)
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/components/WFOForms/AutoFieldLoader.tsx:24]: This is won't fix for the context of this PR

File: packages/orchestrator-ui-components/src/components/confirmationDialog/ConfirmationDialog.tsx [hunk 11]
```diff
@@ -0,0 +1,125 @@
+/*
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/components/confirmationDialog/ConfirmationDialog.tsx:1]: I think this is specific for the temporary solution of forms, maybe move to the WFO Forms folder

File: packages/orchestrator-ui-components/src/contexts/ConfirmationDialogProvider.tsx [hunk 12]
```diff
@@ -0,0 +1,98 @@
+import React from 'react';
```
Comment (ricardovdheijden) [inline packages/orchestrator-ui-components/src/contexts/ConfirmationDialogProvider.tsx:1]: If this is part of the Forms, maybe move to WFOForms folder

**Review by ricardovdheijden — APPROVED** [review]

File: packages/orchestrator-ui-components/src/api/axios.ts (hunk 9 above)
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/api/axios.ts]: True. It was more involved to be able to make it accessible in the app but I have introduced a ApiContextProvider and we are reading it from the context now

File: packages/orchestrator-ui-components/src/api/index.ts (hunk 6 above)
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/api/index.ts:130]: Done

File: packages/orchestrator-ui-components/src/components/WFOForms/CreateForm.tsx (hunk 8 above)
Comment (DutchBen) [reply packages/orchestrator-ui-components/src/components/WFOForms/CreateForm.tsx:38]: Yep

**Review by ricardovdheijden — APPROVED** [review]

**Pull request merged**